ed448/src/field/scalar.rs

```diff
@@ -0,0 +1,1106 @@
+impl From<&Scalar> for Vec<u8> {
+    fn from(scalar: &Scalar) -> Vec<u8> {
+        scalar.to_bytes_rfc_8032().to_vec()
+    }
+}
+
+impl From<Scalar> for ScalarBytes {
+    fn from(scalar: Scalar) -> ScalarBytes {
+        Self::from(&scalar)
+    }
+}
+
+impl From<&Scalar> for ScalarBytes {
+    fn from(scalar: &Scalar) -> ScalarBytes {
+        scalar.to_bytes_rfc_8032()
+    }
+}
+
+#[cfg(any(feature = "alloc", feature = "std"))]
+impl TryFrom<Vec<u8>> for Scalar {
+    type Error = &'static str;
+
+    fn try_from(bytes: Vec<u8>) -> Result<Self, Self::Error> {
+        Self::try_from(&bytes[..])
+    }
+}
+
+#[cfg(any(feature = "alloc", feature = "std"))]
+impl TryFrom<&Vec<u8>> for Scalar {
+    type Error = &'static str;
+
+    fn try_from(bytes: &Vec<u8>) -> Result<Self, Self::Error> {
+        Self::try_from(&bytes[..])
+    }
+}
+
+impl TryFrom<&[u8]> for Scalar {
+    type Error = &'static str;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        if bytes.len() != 57 {
+            return Err("invalid byte length");
+        }
+        let scalar_bytes = ScalarBytes::try_from(bytes).expect("invalid scalar bytes");
+        Option::<Scalar>::from(Scalar::from_canonical_bytes(&scalar_bytes))
+            .ok_or("scalar was not canonically encoded")
+    }
+}
+
+#[cfg(any(feature = "alloc", feature = "std"))]
+impl TryFrom<Box<[u8]>> for Scalar {
+    type Error = &'static str;
+
+    fn try_from(bytes: Box<[u8]>) -> Result<Self, Self::Error> {
+        Self::try_from(bytes.as_ref())
+    }
+}
+
+#[cfg(feature = "serde")]
+impl serdect::serde::Serialize for Scalar {
+    fn serialize<S>(&self, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: serdect::serde::Serializer,
+    {
+        serdect::slice::serialize_hex_lower_or_bin(&self.to_bytes(), s)
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> serdect::serde::Deserialize<'de> for Scalar {
+    fn deserialize<D>(d: D) -> Result<Self, D::Error>
+    where
+        D: serdect::serde::Deserializer<'de>,
+    {
+        let mut buffer = ScalarBytes::default();
+        serdect::array::deserialize_hex_or_bin(&mut buffer[..56], d)?;
+        Option::from(Self::from_canonical_bytes(&buffer)).ok_or(serdect::serde::de::Error::custom(
+            "scalar was not canonically encoded",
+        ))
+    }
+}
+
+#[cfg(feature = "zeroize")]
+impl zeroize::DefaultIsZeroes for Scalar {}
+
+impl core::fmt::LowerHex for Scalar {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        let tmp = self.to_bytes_rfc_8032();
+        for &b in tmp.iter() {
+            write!(f, "{:02x}", b)?;
+        }
+        Ok(())
+    }
+}
+
+impl core::fmt::UpperHex for Scalar {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        let tmp = self.to_bytes_rfc_8032();
+        for &b in tmp.iter() {
+            write!(f, "{:02X}", b)?;
+        }
+        Ok(())
+    }
+}
+
+impl FromOkm for Scalar {
+    type Length = U84;
```

Comment:
This should be 64 bytes as well, see https://github.com/RustCrypto/elliptic-curves/pull/1121#discussion_r2038632770.

Reply:
As pointed out in https://github.com/RustCrypto/elliptic-curves/pull/1121#discussion_r2038632770, this is wrong.